api/include/opentelemetry/logs/logger.h

```diff
@@ -43,81 +60,105 @@ class Logger
+           trace::TraceId trace_id,
+           trace::SpanId span_id,
+           trace::TraceFlags trace_flags) noexcept
   {
-    // Set severity to the default then call log(Severity, String message) method
-    log(Severity::kDefault, message);
+    Log(timestamp, severity, name, body, common::KeyValueIterableView<T>(resources),
+        common::KeyValueIterableView<U>(attributes), trace_id, span_id, trace_flags);
   }
 
-  inline void log(Severity severity, nostd::string_view message) noexcept
+  void Log(core::SystemTimestamp timestamp,
+           Severity severity,
+           nostd::string_view name,
+           nostd::string_view body,
+           std::initializer_list<std::pair<nostd::string_view, common::AttributeValue>> resource,
+           std::initializer_list<std::pair<nostd::string_view, common::AttributeValue>> attributes,
+           trace::TraceId trace_id,
+           trace::SpanId span_id,
+           trace::TraceFlags trace_flags) noexcept
   {
-    // TODO: set default timestamp later (not in API)
-    log(severity, message, core::SystemTimestamp(std::chrono::system_clock::now()));
+    return this->Log(timestamp, severity, name, body,
+                     nostd::span<const std::pair<nostd::string_view, common::AttributeValue>>{
+                         resource.begin(), resource.end()},
+                     nostd::span<const std::pair<nostd::string_view, common::AttributeValue>>{
+                         attributes.begin(), attributes.end()},
+                     trace_id, span_id, trace_flags);
   }
 
-  inline void log(Severity severity,
+  /** Wrapper methods that the user could call for convenience when logging **/
+
+  // Set default values for unspecified fields, then call the base Log() method
+  inline void Log(Severity severity,
                   nostd::string_view message,
                   core::SystemTimestamp time) noexcept
   {
-    // creates a LogRecord object with given parameters, then calls log(LogRecord)
-    LogRecord r;
-    r.severity  = severity;
-    r.body      = message;
-    r.timestamp = time;
-
-    log(r);
+    this->Log(time, severity, default_name_, message, {}, {}, default_trace_id_, default_span_id_,
+              default_trace_flags_);
   }
 
-  /** Overloaded methods for structured logging**/
-  // TODO: separate this method into separate methods since it is not useful for user to create
-  // empty logs
-  template <class T,
-            nostd::enable_if_t<common::detail::is_key_value_iterable<T>::value> * = nullptr>
-  inline void log(Severity severity       = Severity::kDefault,
-                  nostd::string_view name = "",
-                  const T &attributes     = {}) noexcept
+  // Set default time, and call base Log(severity, message, time) method
+  inline void Log(Severity severity, nostd::string_view message) noexcept
   {
-    log(severity, name, common::KeyValueIterableView<T>(attributes));
+    this->Log(severity, message, default_timestamp_);
```

Comment:
I do not have a strong opinion on this. But ideally the `default_timestamp_` should be something like `std::chrono::system_clock::now()`? Maybe we can make an assumption that duration of time `inline` call to `Log` and to the moment when we set it to that actual value (in Exporter?) is rather miniscule.. Basically we log `current name` by default. Maybe an implementation of overridable getter that allows to populate this in concrete SDK logger implementation.

Reply:
+1

Reply:
True, updated this in the latest commit.